tfhe/src/safe_deserialization.rs

```diff
@@ -154,129 +154,109 @@ mod test_shortint {
-                PARAM_MESSAGE_3_CARRY_3_KS_PBS,
-                num_blocks_per_integer
-            ),
+            &params
         )
         .is_err());
 
-        let ct2 = safe_deserialize_conformant::<CompactFheUint8>(
-            buffer.as_slice(),
-            1 << 20,
-            &RadixCiphertextConformanceParams::from_pbs_parameters(
-                PARAM_MESSAGE_2_CARRY_2_KS_PBS,
-                num_blocks_per_integer,
-            ),
-        )
-        .unwrap();
+        let params = CompactFheUint8ConformanceParams::from(&server_key);
+        let ct2 =
+            safe_deserialize_conformant::<CompactFheUint8>(buffer.as_slice(), 1 << 20, &params)
+                .unwrap();
 
         let dec: u8 = ct2.expand().decrypt(&client_key);
         assert_eq!(msg, dec);
     }
 
     #[test]
     fn safe_desererialization_ct_list() {
-        let config = ConfigBuilder::default().build();
-
-        let (client_key, _server_key) = generate_keys(config);
+        let (client_key, server_key) = generate_keys(ConfigBuilder::default().build());
+        let (_, incompatible_server_key) = generate_keys(ConfigBuilder::with_custom_parameters(
+            PARAM_MESSAGE_1_CARRY_1_KS_PBS,
+            None,
+        ));
 
         let public_key = CompactPublicKey::new(&client_key);
 
         let msg = [27u8, 10, 3];
 
-        let num_blocks_per_integer = 4;
-
         let ct_list = CompactFheUint8List::try_encrypt(&msg, &public_key).unwrap();
 
         let mut buffer = vec![];
 
         safe_serialize(&ct_list, &mut buffer, 1 << 40).unwrap();
 
-        let param_set = |list_size_constraint| {
-            RadixCiphertextConformanceParams::from_pbs_parameters(
-                PARAM_MESSAGE_2_CARRY_2_KS_PBS,
-                num_blocks_per_integer,
-            )
-            .to_ct_list_conformance_parameters(list_size_constraint)
-        };
-
-        for parameter_set in [
-            RadixCiphertextConformanceParams::from_pbs_parameters(
-                PARAM_MESSAGE_3_CARRY_3_KS_PBS,
-                num_blocks_per_integer,
-            )
-            .to_ct_list_conformance_parameters(ListSizeConstraint::exact_size(3)),
-            param_set(ListSizeConstraint::exact_size(2)),
-            param_set(ListSizeConstraint::exact_size(4)),
-            param_set(ListSizeConstraint::try_size_in_range(1, 2).unwrap()),
-            param_set(ListSizeConstraint::try_size_in_range(4, 5).unwrap()),
-        ]
-        .iter()
-        {
+        for len_constraint in [
+            ListSizeConstraint::exact_size(3),
+            ListSizeConstraint::exact_size(2),
+            ListSizeConstraint::exact_size(4),
+            ListSizeConstraint::try_size_in_range(1, 2).unwrap(),
+            ListSizeConstraint::try_size_in_range(4, 5).unwrap(),
+        ] {
+            let params = CompactFheUint8ListConformanceParams::from((
+                &incompatible_server_key,
+                len_constraint,
+            ));
```

Comment:
The previous test was testing wrong parameters and wrong size independently
Here, the wrong size is never tested with correct parameters

Reply:
I put back the removed test case